### PR TITLE
python3 compat

### PR DESCRIPTION
**Pull request by flying-sheep**
python3 doesn’t have `exec`

else kudos for your futureproof style!

### CODE REVIEW

**Comment by tonysyu** [comment]
Thanks for the PR!

Are you not seeing the following error when you run `python -m mpl_style_gallery`:

> NameError: name 'np' is not defined

I think this issue is described in [this stackoverflow answer](http://stackoverflow.com/a/29979677/260303). Basically, if the second and third arguments to `exec` are different objects, then global variables inside of `exec`'ed modules are not visible inside functions. By default, those arguments are different (`globals()` and `locals()` dicts). If those two arguments are the same object, which is the case when only the second argument is provided (see second paragraph of [docs](https://docs.python.org/3/library/functions.html#exec)), then this behaves as expected.

In other words, I think `exec(COMMAND)` should become `exec(COMMAND, {})`. That way the empty dictionary is used as the locals and globals dict for the `exec`'d module. For some reason, I wasn't able to come up with a simple example that reproduced the issue, so I may be missing something.

**Comment by flying-sheep** [comment]
done! does it work for you now?

**Comment by tonysyu** [comment]
I'm surprised that you don't see the same issue, but I've checked this out on both python 2.7 and 3.4, so I'm happy to sign off.

Thanks for the contribution!

**Comment by flying-sheep** [comment]
well, i didn’t try with legacy python, so if the error was only there, i didn’t see it.

:smiley:
